Pod/Classes/BONTag.m

```diff
@@ -0,0 +1,235 @@
+            continue;
+        }
+
+        return stringRange;
+    }
+}
+
++ (NSArray *)rangesInString:(NSString **)string betweenTags:(NSArray *)tags stripTags:(BOOL)stripTags
+{
+    NSArray *escapedRanges = [BONTag escapedRangesInString:string withTags:tags stripEscapeCharacters:stripTags];
+
+    NSArray *tagsWithRanges = [[NSArray alloc] initWithArray:tags copyItems:YES];
+
+    NSString *theString = *string;
+
+    NSRange searchRange = NSMakeRange(0, theString.length);
+
+    // Iterate over the string finding each tag in order until there are no more tags
+    while (YES) {
+        BONTag *nextTag;
+        NSRange nextStartTagRange;
+        NSRange nextEndTagRange;
+
+        // Find the next start tag
+        for (BONTag *tag in tagsWithRanges) {
+            NSRange startTagRange = [BONTag findNextString:tag.startTag inString:theString ignoringRanges:escapedRanges range:searchRange];
+            NSRange endTagRange = [BONTag findNextString:tag.endTag inString:theString ignoringRanges:escapedRanges range:searchRange];
+            if (startTagRange.location != NSNotFound && endTagRange.location != NSNotFound) {
+                if (!nextTag || (startTagRange.location < nextStartTagRange.location)) {
+                    nextTag = tag;
+                    nextStartTagRange = startTagRange;
+                    nextEndTagRange = endTagRange;
+                }
+            }
+        }
+
+        if (!nextTag) {
+            break;
+        }
+
+        NSRange range = NSMakeRange(NSMaxRange(nextStartTagRange), nextEndTagRange.location - NSMaxRange(nextStartTagRange));
+
+        if (stripTags) {
+            range.location -= nextTag.startTag.length;
+
+            theString = [theString stringByReplacingOccurrencesOfString:nextTag.startTag withString:@"" options:0 range:nextStartTagRange];
+            nextStartTagRange.length = 0;
+
+            nextEndTagRange.location -= nextTag.startTag.length;
+            theString = [theString stringByReplacingOccurrencesOfString:nextTag.endTag withString:@"" options:0 range:nextEndTagRange];
+            nextEndTagRange.length = 0;
+        }
+
+        [nextTag.ranges addObject:[NSValue valueWithRange:range]];
+
+        searchRange = NSMakeRange(NSMaxRange(nextEndTagRange), [theString length] - NSMaxRange(nextEndTagRange));
+    }
+
+    *string = theString;
+
+    return tagsWithRanges;
+}
+
+#pragma mark - Equality
+
+- (BOOL)isEqualToTag:(BONTag *)tag
+{
+    BOOL startTagEqual = [tag.startTag isEqualToString:self.startTag];
+    BOOL endTagEqual = [tag.endTag isEqualToString:self.endTag];
+    BOOL chainEqual = [tag.textable isEqual:self.textable];
+    BOOL escapeEqual = [tag.escapeString isEqualToString:self.escapeString];
+    BOOL rangesEqual = [tag.ranges isEqualToArray:self.ranges];
+
+    return startTagEqual && endTagEqual && chainEqual && escapeEqual && rangesEqual;
```

Comment:
#nth string these together and order them smartly so you can take advantage of conditional short-circuiting, instead of calculating them all up front.